Assignment04/src/test/java/com/cnu2016/assignment04/Controllers/OrderControllerTest.java

```diff
@@ -0,0 +1,368 @@
+                statusCode(HttpStatus.SC_OK);
+
+        RestAssured.when()
+                .get("/api/orders/{id}", 0).
+                then().
+                statusCode(HttpStatus.SC_NOT_FOUND);
+
+    }
+
+    @Test
+    public void createOrderTest() {
+
+        String orderId =
+            RestAssured.when()
+                .post("/api/orders")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED)
+                .extract().path("id").toString();
+
+//        System.out.println(orderId);
+
+    }
+
+    @Test
+    public void addProductTestWrongFields() {
+
+        String orderId = RestAssured.when()
+                .post("/api/orders")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED)
+                .extract().path("id").toString();
+        Integer id = Integer.parseInt(orderId);
+
+        /* Not giving the specified fields */
+        RestAssured.given()
+                .contentType("application/json")
+                .body("{}")
+                .when()
+                .post("/api/orders/{id}/orderLineItem", id)
+                .then()
+                .statusCode(HttpStatus.SC_BAD_REQUEST);
+    }
+
+    @Test
+    public void addProductTestMoreThanStock() {
+
+        /* Adding more quantity than in stock */
+
+        String orderId = RestAssured.when()
+                .post("/api/orders")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED)
+                .extract().path("id").toString();
+
+        Integer id = Integer.parseInt(orderId);
+
+        product tempProduct = new product();
+        tempProduct.setQuantityInStock(99);
+        tempProduct = productRepository.save(tempProduct);
+        Integer productId = tempProduct.getId();
+
+        RestAssured.given()
+                .contentType("application/json")
+                .body("{\"product_id\":" + productId + ",\"qty\":\"100\"}")
+                .when()
+                .post("/api/orders/{id}/orderLineItem", id)
+                .then()
+                .statusCode(HttpStatus.SC_CREATED);
+    }
+
+    @Test
+    public void addProductTestWrongIdCheck() {
+
+        /* Wrong product_id check */
+
+        String orderId = RestAssured.when()
+                .post("/api/orders")
+                .then()
+                .statusCode(HttpStatus.SC_CREATED)
+                .extract().path("id").toString();
+
+        Integer id = Integer.parseInt(orderId);
+        product tempProduct = new product();
+        tempProduct.setQuantityInStock(99);
+        tempProduct = productRepository.save(tempProduct);
+        Integer productId = tempProduct.getId();
+
+        RestAssured.given()
+                .contentType("application/json")
+                .body("{\"product_id\":\"wrongInput\",\"qty\":\"100\"}")
+                .when()
+                .post("/api/orders/{id}/orderLineItem", id)
+                .then()
+                .statusCode(HttpStatus.SC_BAD_REQUEST);
+    }
```

Comment:
using other api's inside test of the api, could go it separately using repo in functions, so as to reuse code